mono/utils/mono-threads-state-machine.c

```diff
@@ -197,7 +194,7 @@ mono_threads_transition_attach (MonoThreadInfo* info)
 			mono_fatal_with_history ("suspend_count = %d, but should be == 0", suspend_count);
 		if (no_safepoints)
 			mono_fatal_with_history ("no_safepoints = TRUE, but should be FALSE");
-		if (thread_state_cas (&info->thread_state, STATE_RUNNING, raw_state) != raw_state)
+		if (thread_state_cas (&info->thread_state, build_thread_state (STATE_RUNNING, 0, 0), raw_state) != raw_state)
```

Comment:
These and similar are the importane lines.

Reply:
nit: the third argument to `build_thread_state` should be `FALSE`, not `0`.